### DISCUSSION TITLE
Incorrect value (0) for custom primary keys without auto-increment

### DISCUSSION THREAD

**Discussion by Gredys** [adonisjs/core#3251] [Help] [answered]
## Sample Code (to reproduce the issue)
Any model with custom `primaryKey` column
```
import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'

export default class Block extends BaseModel {
  @column({ isPrimary: true })
  public number: number

  @column()
  public other: number
}
```
Sample migration for such model
```
import BaseSchema from '@ioc:Adonis/Lucid/Schema'

export default class Blocks extends BaseSchema {
  protected tableName = 'blocks'

  public async up () {
    this.schema.createTable(this.tableName, (table) => {
      table.integer('number')
      table.integer('other')
    })
  }

  public async down () {
    this.schema.dropTable(this.tableName)
  }
}
```

## Description of problem
Creating model with custom primary key and retriving of that primary key lead to incorrect values.
Example of input and output data below.

Example input for create: 
```
{
   number: 1337,
   other: 1337
}
```

Example output of model of create call:
```
--> Block {
  modelOptions: undefined,
  modelTrx: undefined,
  transactionListener: [Function: bound listener],
  fillInvoked: true,
  cachedGetters: {},
  '$columns': {},
  '$attributes': {
    number: 0,
    other: 1337
  },
  '$original': {
    number: 0,
    number: 1337,
  },
  '$preloaded': {},
  '$extras': { id: 0 },
  '$sideloaded': {},
  '$isPersisted': true,
  '$isDeleted': false,
  '$isLocal': true,
  number: 1337,
  other: 1337
}
```

Example output of model after toJSON call:
```
{
    number: 0,
    other: 1337
}
```

Saved data in db contains correct values regadless of output in code.

## Temporary solution (don't think its right)
Removing `isPrimary` attrbitue from column fixes completely this issue. Returned data after `create` and `toJSON` calls cotains corrent value in `number` column/field.

## Question
How to deal with such situations? How correctly create custom primary keys are retrive its real value instead of `0`?

**Comment by thetutlage** [comment, marked as answer]
Since you are self assigning the primary key in the application codebse. You need to use this property to notify Lucid https://docs.adonisjs.com/reference/orm/base-model#static-selfassignprimarykey

**Comment by Gredys** [comment]
So, its not enought just to mark another column with `isPrimary`? And `selfAssignPrimaryKey` is needed to define custom `primaryKey`?
